Выполнено ДЗ 13го спринта

IKolmogorov opened the pull request:
Передаю ДЗ 13го спринта на проверку.

IKolmogorov committed:
Выполнено ДЗ 13го спринта

IliaAnK commented on app/src/main/java/com/example/playlistmaker2/AudioPlayerActivity.kt:
```diff
@@ -0,0 +1,77 @@
+        binding.apply {
+
+            if (curTrack != null) {
+                artistName.setText(curTrack?.artistName.toString())
```
Рекомендация:
`artistName.setText(curTrack?.artistName.toString())` -> `artistName.text = curTrack.artistName`
Ниже аналогично

IliaAnK commented on app/src/main/java/com/example/playlistmaker2/AudioPlayerActivity.kt:
```diff
@@ -0,0 +1,77 @@
+
+        binding = ActivityAudioPlayerBinding.inflate(layoutInflater)
+
+        val curTrackString = intent.getStringExtra("curTrack")
```
Рекомендация:
Нет смысла хранить ссылки на `View` в полях класса, когда есть `ViewBinding`.
С ним можно обращаться к `View` напрямую
Для остальных экранов **аналогично**.

IKolmogorov committed:
Спринт13. Проведена работа над ошибками:

• Критические замечания:

	1) У всех экранов, кроме экрана "Аудиоплеер", в тёмной теме цвет background’а не соответствует макетам. Согласно макетам он должен быть YP_Black(#1A1B22) цвета;
		Решение: Создал новый ресур типа color с названием YP_Black. Для аттрибутов "mainActivityBackgroundColor" и "otherActivityBackgroundColor" изменил значение в themes (night) с "Black" на "YP_Black". Проверил поиском по проекту, что аттрибуты используются только для backgroud основных View в Activities. Для экрана "Аудиоплеер" ввел стиль и также теперь использую аттрибут "otherActivityBackgroundColor".

	2) У экрана "Аудиоплеер", в светлой теме цвет background’а не соответствует макетам;
		Решение: Решилось параллельно с решением пункта "1".

	3) Баг. Окружение: эмулятор Pixel 6, 33 Api
Кейс: перейти на экран "Поиск" -> выполнить поиск -> появился результат поиска(список треков/заглушка "Нет результатов" или "Нет интернета") -> стереть запрос с помощью клавиатуры
ОР: последний результат поиска скрывается, если история не пуста, то появляется история
ФР: последний результат поиска остался на экране.

		Решение: Доработал обработчик afterTextChanged TextWatchera для поля ввода поискового запроса

• Рекомендации:

	1) Баг. Окружение: эмулятор Pixel 6, 33 Api, на устройстве включена тёмная тема
Кейс: установить и открыть приложение
ОР: включена тёмная тема
ФР: включена светлая тема;

	2) Баг. Окружение: эмулятор Pixel 6, 33 Api, на устройстве включена тёмная тема
Кейс: установить и открыть приложение -> перейти на экран «Настройки»
ОР: свичер тёмной темы включен
ФР: свичер тёмной темы выключен

	Решение 1,2: Сделал все по предложенному алгоритму. При запуске приложения проверяю, что в SharedPreferences есть значение по идентификатору DARK_THEME и, если нет, то с пощью созданнй функции определяю текущее значений системной темы и записываю его в SharedPreferences. Далее работает с помощью разработанного ранее функционала.

 • Рекомендации с GitHub-а:

	1) artistName.setText(curTrack?.artistName.toString()) -> artistName.text = curTrack.artistName. Ниже аналогично

	Решение: Исправлено в соответствии с рекомендацией.

	2) Нет смысла хранить ссылки на View в полях класса, когда есть ViewBinding. С ним можно обращаться к View напрямую. Для остальных экранов аналогично.

	Решение: Не до конца разобрался в этом вопросе. Насколько я понял, в данном случае мне достаточно просто getExtra из Intent и не требуется взаимодействия с ViewBinding. Внес правки - убрал лишний код.
